Add options to extend list of sensitive keywords

Marloxo opened the pull request:
---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Target Github ISSUE in description if exists
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

this PR addressed the following issue: https://github.com/apache/airflow/issues/9372

basically it's adding support for additional labels from the configuration so users can extend the matching of sensitive keywords in variables name.

Marloxo committed:
add sensitive_variable_fields

mik-laj reviewed (approved):
Can you add tests to avoid regression?

Marloxo committed:
Add test and fix DEFAULT_SENSITIVE_VARIABLE_FIELDS type

change DEFAULT_SENSITIVE_VARIABLE_FIELDS to set instead of tuple
add tests for SENSITIVE_VARIABLE_FIELDS

mik-laj commented on airflow/config_templates/config.yml:
```diff
@@ -1678,6 +1678,13 @@
       default: "True"
+    - name: sensitive_variable_fields
+      description: |
+        Sensitive keywords to look for in variables names
```
```suggestion
        A comma-separated list of sensitive keywords to look for in variables names.
```

mik-laj commented on airflow/www/utils.py:
```diff
@@ -39,23 +39,28 @@
+        sensitive_variable_fields = conf.get('admin', 'sensitive_variable_fields')
+
+        if sensitive_variable_fields:
+            DEFAULT_SENSITIVE_VARIABLE_FIELDS.update(sensitive_variable_fields.split(','))
```
This will cause this variable to be updated each time this method is called. I think it's enough for this to happen only once.

Marloxo replied:
Good point

Marloxo committed:
apply suggestions

Marloxo committed:
add doc to security

mik-laj commented on docs/security.rst:
```diff
@@ -421,3 +421,15 @@ the new key to the ``fernet_key`` setting, run
+-------------------------
+
+By default, Airflow Value of a variable will be hidden if the key contains any words in
+(‘password’, ‘secret’, ‘passwd’, ‘authorization’, ‘api_key’, ‘apikey’, ‘access_token’), but can be configured
```
Can you add description for hide_sensitive_variable_fields option here also?

Marloxo replied:
Sure, updated

Marloxo committed:
add description in UI doc page
